Palindrome.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+
+namespace Palindrome
+{
+    public sealed class Palindrome
+    {
+        /// <summary>
+        /// Method checks if input string is a palindrome.
+        /// Altered method to accept string input instead of char Array.
+        /// Functionally they are equivalent.
+        /// Modified to execute without duplicating strings using 
+        /// minimum space complexity.
+        /// </summary>
+        /// <param name="arr">A character array.</param>
+        /// <returns>Boolean result</returns>
+        public bool IsPalindrome(string strInput)
+        {
+            // Defines empty input as NOT a palindrome.
+            if (String.IsNullOrEmpty(strInput))
+            {
+                throw new ArgumentNullException(nameof(strInput));
+            }
+
+            int left = 0;
+            int right = strInput.Length - 1;
+
+            while (left < right)
+            {
+                // Skip over spaces to allow for sentences.
+                if (strInput[left] == ' ')
+                {
+                    left++;
```

Comment:
You should add a test case for a string "bob " (notice the space at the end). I think you'll need a "continue;" here.